pkg/controlplane/instance.go

```diff
@@ -540,6 +566,9 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 			leaseName := m.GenericAPIServer.APIServerID
 			holderIdentity := m.GenericAPIServer.APIServerID + "_" + string(uuid.NewUUID())
 
+			peeraddress := getPeerAddress(c.GenericConfig.PeerAdvertiseAddress, c.GenericConfig.PublicAddress, publicServicePort)
+			// must replace ':' in <ip:port> to be able to store this as a valid label value
```

Comment:
are brackets a possible problem too `[` ?

Reply:
updated to replace `[]` as well.